backend/src/Service/Admin/Order/AdminOrderService.php

```diff
@@ -1954,4 +1960,127 @@ public function getCaptainsWhoDeliveredLastFiveOrders(): array
+            $orderCreateByAdminRequest->setOrderCost(OrderCostDefaultValueConstant::ORDER_VALUE_CONST);
+            $orderCreateByAdminRequest->setNote(OrderNoteDefaultValueConstant::ORDER_NOTE_DEFAULT_VALUE_CONST);
+            $orderCreateByAdminRequest->setDestination(OrderClientLocationConstant::ORDER_CLIENT_LOCATION_ARRAY_CONST[array_rand(OrderClientLocationConstant::ORDER_CLIENT_LOCATION_ARRAY_CONST)]);
+            $orderCreateByAdminRequest->setRecipientName(OrderRecipientDefaultValueConstant::ORDER_RECIPIENT_NAME_DEFAULT_VALUE_CONST);
+            $orderCreateByAdminRequest->setRecipientPhone(OrderRecipientDefaultValueConstant::ORDER_RECIPIENT_PHONE_DEFAULT_VALUE_CONST);
+            $orderCreateByAdminRequest->setDetail(OrderNoteDefaultValueConstant::ORDER_NOTE_DEFAULT_VALUE_CONST);
+
+            // 6 calculate the distance between the store's branch and the client location
+            $distance = $this->getDistanceBetweenTwoLocations($branch->getLocation(), $orderCreateByAdminRequest->getDestination());
+            $orderCreateByAdminRequest->setStoreBranchToClientDistance($distance);
+
+            // 7 calculate the delivery cost
+            $orderCreateByAdminRequest->setDeliveryCost($this->calculateDeliveryCostByStoreOwnerProfileIdAndDistance($storeOwnerProfile->getId(),
+                $distance));
+
+            // 8 create order record
+            $order = $this->adminOrderManager->createOrderByAdmin($orderCreateByAdminRequest);
+
+            if ($order) {
+                // 9 update the remaining orders of the store's subscription
+                $this->updateRemainingOrdersOfStoreSubscriptionByAdmin($storeOwnerProfile->getId(), $order->getCreatedAt(),
+                    SubscriptionConstant::OPERATION_TYPE_SUBTRACTION, 1);
+
+                // 10 create local notification for the store
+                $this->createLocalNotificationForStore($storeOwnerProfile->getId(), NotificationConstant::NEW_ORDER_TITLE,
+                    NotificationConstant::CREATE_ORDER_SUCCESS, $order->getId());
+
+                // 11 create order timeline
+                $this->createOrderLogViaOrderEntity($order);
+
+                // 12 save log of the action on order
+                $this->createOrderLogMessageViaOrderEntityAndByAdmin($order, $userId);
+
+                // 13 prepare the response
+                $response[] = $this->autoMapping->map(OrderEntity::class, TestingOrderCreateByAdminResponse::class, $order);
+
+                // 14 create firebase notification to store
+                $this->sendFirebaseNotificationAboutOrderStateForUserByAdmin($storeOwnerProfile->getStoreOwnerId(),
+                    $order->getId(), $order->getState(), NotificationConstant::STORE);
+
+                // 15 create firebase notification to captains
+                $this->sendFirebaseNotificationToAllCaptainsByAdmin($order->getId());
+            }
+        }
+
+        return $response;
+    }
```

Comment:
This is very sad, please move everything related to this craziness to a new service.